### ISSUE TITLE
[CT-216] Replace `MainReportArgs` event with `MainReportFlags`

### ISSUE THREAD

**Issue by jtcohen6** [dbt-labs/dbt-core#4709]
Now that we've normalized ["global configs"](https://docs.getdbt.com/reference/global-configs), which can be set in any of three places (CLI flag, env var, `profiles.yml` user config)...

We should consider swapping `MainReportArgs` with `MainReportFlags`, and report the values of [`flags.get_flag_dict()`](https://github.com/dbt-labs/dbt-core/blob/434f3d26789d4bed3f0057c7c7d898e9dbf6687f/core/dbt/flags.py#L152) instead of just `argparse.Namespace`:

https://github.com/dbt-labs/dbt-core/blob/434f3d26789d4bed3f0057c7c7d898e9dbf6687f/core/dbt/main.py#L233

https://github.com/dbt-labs/dbt-core/blob/434f3d26789d4bed3f0057c7c7d898e9dbf6687f/core/dbt/events/types.py#L148-L153

Consider the following:
```
$ DBT_DEBUG=True dbt run


============================== 2022-02-10 16:45:30.794086 | 5b29fa3d-7fb4-49db-938f-09ed631528f6 ==============================
16:45:30.794086 [info ] [MainThread]: Running with dbt=1.0.1
16:45:30.795971 [debug] [MainThread]: running dbt with arguments Namespace(record_timing_info=None, debug=None, log_format=None, write_json=None, use_colors=None, printer_width=None, warn_error=None, version_check=None, partial_parse=None, single_threaded=False, use_experimental_parser=None, static_parser=None, profiles_dir='/Users/jerco/.dbt', send_anonymous_usage_stats=None, fail_fast=None, event_buffer_size=None, project_dir=None, profile=None, target=None, vars='{}', log_cache_events=False, threads=None, select=None, exclude=None, selector_name=None, state=None, defer=None, full_refresh=False, cls=<class 'dbt.task.run.RunTask'>, which='run', rpc_method='run')
```
We're _clearly_ using `--debug` mode here, but because we set this via _env var_ instead of _CLI flag/arg_, the event reports `debug=None`. This confused us when trying to debug (hah) a global config earlier.

It's sorta helpful to know which values the user passes as CLI flags in particular, but it's much more useful to know the values of each config that's actually used in the invocation.

We might also think about:
- Serializing the output of this event to a JSON dict, when JSON formatted. (That might come for free if we use the flag dict, instead of the `Namespace` object.)
- Saving `flags` instead of `args` in the `RunResults` artifact ([here](https://github.com/dbt-labs/dbt-core/blob/41c6177ae2c85343426c5af53f9e23451e762619/core/dbt/contracts/results.py#L198) + [here](https://github.com/dbt-labs/dbt-core/blob/41c6177ae2c85343426c5af53f9e23451e762619/core/dbt/task/runnable.py#L593))

**Comment by gshank** [comment]
Including the flags was fixed in main when I did the mashumaro conversion. Currently we have fire_event(MainReportArgs(args=args_to_dict(parsed))), and that 'args_to_dict' method in core/dbt/utils.py includes the flags.get_flag_dict(). That utility was already used in the RunResult, and for consistency I also used in in MainReportArgs.

**Comment by jtcohen6** [comment]
Amazing!

https://github.com/dbt-labs/dbt-core/blob/41c6177ae2c85343426c5af53f9e23451e762619/core/dbt/utils.py#L630-L635

Closing as resolved in `main`, for inclusion in v1.1 :)
